.github/workflows/build-test-publish-images.yml

```diff
@@ -228,10 +250,11 @@ jobs:
         ${{ needs.compute-matrix.outputs.RAPIDS_VER }}\
         ${{ needs.compute-matrix.outputs.ALPHA_TAG }}-\
         cuda${{ matrix.CUDA_VER }}-\
-        py${{ matrix.PYTHON_VER }}"
+        py${{ matrix.PYTHON_VER }}-\
+        ${{ matrix.ARCH }}"
   delete-temp-images:
     if: ${{ !cancelled() && needs.test.result == 'success' }}
-    needs: [compute-matrix, build-multiarch-manifest, test]
+    needs: [compute-matrix, build, test]
```

Comment:
I can help explain.

`build` publishes images with tags like this:

```text
rapidsai/staging:docker-notebooks-706-24.10a-cuda12.0-py3.11-arm64
rapidsai/staging:docker-notebooks-706-24.10a-cuda12.0-py3.11-amd64
```

The `build-multiarch-manifest` job pushes a new manifest to DockerHub which combines those 2 images together, so that something like this will work:

```shell
docker pull \
    rapidsai/staging:docker-notebooks-706-24.10a-cuda12.0-py3.11
```

*(notice .... no `-{arch}` suffix)*

As of #702, that `build-multiarch-manifest` job is not run on pull requests at all.

https://github.com/rapidsai/docker/blob/1c27d9245fd9d99ee35981b970acaf10961ca45b/.github/workflows/build-test-publish-images.yml#L171-L172

I believe that was intentional, as a way to cut down on CI time and network calls. The combined manifests are a convenience for downstream consumers of these images, and not necessary for one CI job here to read the outputs from a prior job.

Reply:
Thanks James! 🙏

That's a helpful explanation. Agree we don't want this constraint in PRs

Though I think with builds on branches, this was a change that Jake introduced in PR ( https://github.com/rapidsai/docker/pull/702 ) to make sure we were not deleting images before the manifest was generated

https://github.com/rapidsai/docker/blob/1c27d9245fd9d99ee35981b970acaf10961ca45b/.github/workflows/build-test-publish-images.yml#L232-L234

Otherwise if a build failed, we would delete the images needed for the manifest and restarting the failed jobs would just fail immediately (as the manifest cannot be created)

Reply:
gah you're 100% right. We need to account for that here. I'll push some changes.

That condition never could have worked on PRs, because `build-multiarch-manifest` doesn't run on PRs (see related conversation in this other thread: https://github.com/rapidsai/docker/pull/708#discussion_r1718781188). Sorry I didn't catch it in previous reviews.